src/sentry/pipeline/__init__.py

```diff
@@ -33,7 +37,13 @@ def set_config(self, config):
         the provider instance. This is useful for example when nesting
         pipelines and the provider needs to be configured differently.
         """
-        self.config = config
+        try:
```

Comment:
Yeah I think if we just ensure config is always set, we can get rid of this logic.

We should probably rename `set_config` to `update_config` too then